src/cozmo_companion/assistant.py

```diff
@@ -103,47 +127,82 @@ def _listen(self):
                     keywords=KEYWORDS,
                     keywords_threshold=KEYWORDS_THRESHOLD,
                 ).get_result()
-
+                # Extract the transcribed text from the result
                 user_speech_text = speech_result["results"][0]["alternatives"][0][
                     "transcript"
                 ]
 
             return user_speech_text
+        # Handle exceptions from the IBM service
         except ApiException as ex:
             logging.error(
                 "Method failed with status code " + str(ex.code) + ": " + ex.message
             )
 
+    # @tool
+    def get_feedback_inquiry(self, user_text: str = "") -> Optional[str]:
+        """
+        Checks if the user's text contains specific feedback phrases and
+        returns an appropriate inquiry string based on the content.
```

Comment:
Create follow-up issue to generalize this function and check whether it contains any feedback question (using LLM)

Reply:
Issue created.